libs/pika/async_cuda/tests/unit/then_with_stream.cu

```diff
@@ -113,11 +119,13 @@ struct dummy
         return x + 1;
     }
 
+#if !defined(PIKA_HAVE_HIP)
     int operator()(cusolverDnHandle_t, double x) const
     {
         ++cusolver_double_calls;
         return x + 1;
     }
+#endif
```

Comment:
With this change the expected call counts will need to be updated for HIP since the `cusolverDnHandle_t` overloads will never be called.

Reply:
Thanks, yes I went too fast